From: SeongJae Park <sj@kernel.org>
Date: Sun, 27 Jul 2025 10:39:21 -0700
Subject: [PATCH] mm/mprotect: use PAGE_NONE only if
 ARCH_SUPPORTS_NUMA_BALANCING

On !ARCH_SUPPORTS_NUMA_BALANCING, PAGE_NONE is not defined.

Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202507270855.YvsYlfrv-lkp@intel.com/
Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/mprotect.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/mprotect.c b/mm/mprotect.c
index 8b830755fec1..56c61b7d0d2f 100644
--- a/mm/mprotect.c
+++ b/mm/mprotect.c
@@ -714,7 +714,7 @@ long change_protection(struct mmu_gather *tlb,
 	WARN_ON_ONCE(cp_flags & MM_CP_PROT_NUMA);
 #endif
 
-#ifdef CONFIG_DAMON
+#ifdef ARCH_SUPPORTS_NUMA_BALANCING
 	if (cp_flags & MM_CP_DAMON)
 		newprot = PAGE_NONE;
 #endif
-- 
2.39.5

